policy/task/trusted_artifacts.rego

```diff
@@ -59,3 +79,14 @@ _ta_results contains result_name if {
 _has_ta_suffix(name) if endswith(name, "_ARTIFACT")
 
 _is_ta_step(step) if contains(step.image, "trusted-artifacts")
+
+_uses_trusted_artifacts(task) if {
```

Comment:
I'm wondering how bulletproof this detection method is. Could it produce false positives? Maybe add some commentary here to explain the assumptions being made and if/how it might reach the wrong conclusion.

Reply:
Oh, I realized we're checking a task definitions not attestation after writing that. So it's probably no big deal. Even so some comments would be useful.

Reply:
Good idea. Added a comment explaining the approach.